src/pybind/mgr/dashboard/frontend/package.json

```
@@ -75,15 +75,15 @@
     "testURL": "http://localhost/"
   },
   "dependencies": {
-    "@angular/animations": "11.2.14",
-    "@angular/common": "11.2.14",
-    "@angular/compiler": "11.2.14",
-    "@angular/core": "11.2.14",
-    "@angular/forms": "11.2.14",
-    "@angular/localize": "11.2.14",
-    "@angular/platform-browser": "11.2.14",
-    "@angular/platform-browser-dynamic": "11.2.14",
-    "@angular/router": "11.2.14",
+    "@angular/animations": "12.2.8",
+    "@angular/common": "12.2.8",
+    "@angular/compiler": "12.2.8",
+    "@angular/core": "12.2.8",
+    "@angular/forms": "12.2.8",
+    "@angular/localize": "12.2.8",
+    "@angular/platform-browser": "12.2.8",
+    "@angular/platform-browser-dynamic": "12.2.8",
+    "@angular/router": "12.2.8",
     "@circlon/angular-tree-component": "10.0.0",
     "@ng-bootstrap/ng-bootstrap": "7.0.0",
```

```suggestion
    "@ng-bootstrap/ng-bootstrap": "10.0.0",
```
It seems we need to [upgrade the version](https://github.com/ng-bootstrap/ng-bootstrap/releases/tag/10.0.0).

---

Ah yes, i'll take care of that here. Also would it make sense to go through most of the npm packages and upgrade them (not everything but atleast some of the important one) in a different PR? @alfonsomthd

---

In this PR we should upgrade only those that are required in order to work properly with Angular 12.

---

Done and created a [tracker for npm upgrades ](https://tracker.ceph.com/issues/53274)

---

ng-bootstrap is causing some issues with unit-testing, so I'll fix that and update

---

Done!